ui/app/components/dashboard/client-count-card.js

```diff
@@ -4,50 +4,52 @@
  */
 
 import Component from '@glimmer/component';
-import getStorage from 'vault/lib/token-storage';
 import timestamp from 'core/utils/timestamp';
 import { task } from 'ember-concurrency';
 import { waitFor } from '@ember/test-waiters';
 import { tracked } from '@glimmer/tracking';
 import { service } from '@ember/service';
+import { setStartTimeQuery } from 'core/utils/client-count-utils';
 
 /**
  * @module DashboardClientCountCard
  * DashboardClientCountCard component are used to display total and new client count information
  *
  * @example
- * ```js
- * <Dashboard::ClientCountCard @license={{@model.license}}  />
- * ```
- *  @param {object} license - license object passed from the parent
+ *
+ * <Dashboard::ClientCountCard />
+ *
  */
 
 export default class DashboardClientCountCard extends Component {
   @service store;
 
+  clientConfig = null;
+  licenseStartTime = null;
   @tracked activityData = null;
-  @tracked clientConfig = null;
   @tracked updatedAt = timestamp.now().toISOString();
 
   constructor() {
     super(...arguments);
     this.fetchClientActivity.perform();
-    this.clientConfig = this.store.queryRecord('clients/config', {}).catch(() => {});
   }
 
   get currentMonthActivityTotalCount() {
     return this.activityData?.byMonth?.lastObject?.new_clients.clients;
   }
 
-  get licenseStartTime() {
-    return this.args.license.startTime || getStorage().getItem('vault:ui-inputted-start-date') || null;
-  }
-
   @task
   @waitFor
   *fetchClientActivity(e) {
     if (e) e.preventDefault();
     this.updatedAt = timestamp.now().toISOString();
+
+    if (!this.clientConfig) {
+      // set config and license start time when component initializes
+      this.clientConfig = yield this.store.queryRecord('clients/config', {}).catch(() => {});
+      this.licenseStartTime = setStartTimeQuery(this.args.isEnterprise, this.clientConfig);
```

Comment:
technically I could just pass `true` here instead of `this.args.isEnterprise` because this component is wrapped in a conditional and only renders for enterprise versions. however, in case a future refactor shuffled conditionals, I wanted to preserve the logic in `setStartTimeQuery` which prevents from accidentally setting the billing start time to `0001-01-01T00:00:00Z` for community versions